src/ui/public/notify/__tests__/notifier.js

```diff
@@ -43,8 +51,47 @@ describe('Notifier', function () {
       expect(notify('error').title).to.equal('Error');
     });
 
-    it('sets lifetime to Infinity', function () {
-      expect(notify('error').lifetime).to.equal(Infinity);
+    it('sets lifetime to 5 minutes', function () {
+      expect(notify('error').lifetime).to.equal(300000);
+    });
+
+    it('sets timeRemaining and deprecates', function () {
+      var notif = notify('error');
+
+      expect(notif.timeRemaining).to.equal(300);
+      $interval.flush(1000);
```

Comment:
Do you have experience doing this sort of test before?  I worry that this introduces a timing issue since there's no guarantee in javascript that a timeout will actually occur after exactly the duration given.  Then again, it's entirely possible that the mock interval service just tracks all of these durations as flat values, so it could be a non-issue...